sql/types/number.go

```diff
@@ -217,101 +226,93 @@ func (t NumberTypeImpl_) Convert(v interface{}) (interface{}, error) {
 	if jv, ok := v.(JSONValue); ok {
 		jd, err := jv.Unmarshall(nil)
 		if err != nil {
-			return nil, err
+			return nil, sql.InRange, err
 		}
 		v = jd.Val
 	}
 
 	switch t.baseType {
 	case sqltypes.Int8:
-		num, err := convertToInt64(t, v)
+		num, _, err := convertToInt64(t, v)
 		if err != nil {
-			return nil, err
+			return nil, sql.InRange, err
 		}
-		if num > math.MaxInt8 || num < math.MinInt8 {
-			return nil, sql.ErrValueOutOfRange.New(num, t)
+		if num > math.MaxInt8 {
+			return int8(math.MaxInt8), sql.OutOfRange, nil
+		} else if num < math.MinInt8 {
+			return int8(math.MinInt8), sql.OutOfRange, nil
 		}
-		return int8(num), nil
+		return int8(num), sql.InRange, nil
 	case sqltypes.Uint8:
-		num, err := convertToUint64(t, v)
-		if err != nil {
-			return nil, err
-		}
-		if num > math.MaxUint8 {
-			return nil, sql.ErrValueOutOfRange.New(num, t)
-		}
-		return uint8(num), nil
+		return convertToUint8(t, v)
 	case sqltypes.Int16:
-		num, err := convertToInt64(t, v)
+		num, inRange, err := convertToInt64(t, v)
```

Comment:
You aren't consistent in this function about whether you use inRange for the return value, or not. Seems like you probably want `_` here and use `sql.InRange` in the normal return statement